packages/amplify-cli/src/execution-manager.ts

```diff
@@ -140,64 +141,94 @@ async function selectPluginForExecution(context: Context, pluginCandidates: Plug
   return result;
 }
 
-async function executePluginModuleCommand(context: Context, plugin: PluginInfo) {
+async function executePluginModuleCommand(context: Context, plugin: PluginInfo): Promise<void> {
   const { commands, commandAliases } = plugin.manifest;
   if (!commands!.includes(context.input.command!)) {
     context.input.command = commandAliases![context.input.command!];
   }
 
-  if (fs.existsSync(plugin.packageLocation)) {
-    await raisePreEvent(context);
+  if (!fs.existsSync(plugin.packageLocation)) {
+    await scan();
+    context.print.error('The Amplify CLI plugin platform detected an error.');
+    context.print.info('It has performed a fresh scan.');
+    context.print.info('Please execute your command again.');
+    return;
+  }
 
-    const pluginModule = require(plugin.packageLocation);
-    if (
-      pluginModule.hasOwnProperty(constants.ExecuteAmplifyCommand) &&
-      typeof pluginModule[constants.ExecuteAmplifyCommand] === 'function'
-    ) {
-      attachContextExtensions(context, plugin);
-      await pluginModule.executeAmplifyCommand(context);
-    } else {
-      // if the module does not have the executeAmplifyCommand method,
-      // fall back to the old approach by scanning the command folder and locate the command file
-      let commandFilepath = path.normalize(path.join(plugin.packageLocation, 'commands', plugin.manifest.name, context.input.command!));
-      if (context.input.subCommands && context.input.subCommands.length > 0) {
-        commandFilepath = path.join(commandFilepath, ...context.input.subCommands!);
-      }
+  let commandName = constants.ExecuteAmplifyCommand;
+  let fallbackFn = legacyCommandExecutor;
 
-      let commandModule;
+  if (isHeadlessCommand(context)) {
+    commandName = constants.ExecuteAmplifyHeadlessCommand;
+    fallbackFn = () => context.print.error(`Headless mode is not implemented for ${plugin.packageName}`);
+  }
 
-      try {
-        commandModule = require(commandFilepath);
-      } catch (e) {
-        // do nothing
-      }
+  return executionHandler(context, plugin)
```

Comment:
Not able to understand the need of having this chain with the "forCommand" out here. Why can't the logic just be inside the executionHandler?

Reply:
Yeah probably would be cleaner to just move the headless check into the executionHandler